src/execution/__tests__/executor-test.js

```diff
@@ -1029,4 +1030,42 @@ describe('Execute: Handles basic execution tasks', () => {
     const result = execute({ schema, document, fieldResolver });
     expect(result).to.deep.equal({ data: { foo: 'foo' } });
   });
+
+  it('uses a custom type resolver', () => {
+    const document = parse('{ foo { bar } }');
+
+    const fooInterface = new GraphQLInterfaceType({
+      name: 'FooInterface',
+      fields: {
+        bar: { type: GraphQLString },
+      },
+    });
+
+    const fooObject = new GraphQLObjectType({
+      name: 'FooObject',
+      interfaces: [fooInterface],
+      fields: {
+        bar: { type: GraphQLString },
+      },
+    });
+
+    const schema = new GraphQLSchema({
+      query: new GraphQLObjectType({
+        name: 'Query',
+        fields: {
+          foo: { type: fooInterface },
+        },
+      }),
+      types: [fooObject],
+    });
+
+    function typeResolver(source, context, info, abstractType) {
+      return schema.getPossibleTypes(abstractType)[0];
```

Comment:
Can we change this line to something like `return schema.getPossibleTypes(abstractType).find(graphQLType => graphQLType.name === "FooObject");`. Makes it clearer what we are returning.

Reply:
@Neitsch Great suggestion 👍
One thing is that it's enough to return string with type name from `typeResolve` so matching type by name looks unnecessary if you can just return name.

So I just change it to be more explicit.